torch/library.py

```diff
@@ -351,7 +379,7 @@ def impl_abstract(qualname, func=None, *, lib=None, _stacklevel=1):
         >>>     result = x.new_empty(shape, dtype=torch.int64)
         >>>     return result
         >>>
-        >>> @torch.library.impl(lib, "custom_nonzero", "CPU")
+        >>> @torch.library.impl("mylib::custom_nonzero", "cpu")
```

Comment:
With the new API, would it be best practice to omit passing lib all the time for simplicity, and rely on impl/define functions to create fragments behind the scenes? If that is the case should we accept the lib parameter at all, or are there other reasons for having it around?

Reply:
Could another way be just to have some kind of global mapping somewhere that remembers how to map ns to lib?

Reply:
> With the new API, would it be best practice to omit passing lib all the time for simplicity, and rely on impl/define functions to create fragments behind the scenes?

Yup

>  If that is the case should we accept the lib parameter at all, or are there other reasons for having it around?

This is useful for testing. I don't expect anyone to actually use the lib parameter, but I'm keeping it around as an option.

Reply:
I see, thanks.